CRM/Contribute/Form/Contribution.php

```diff
@@ -1501,6 +1507,21 @@ protected function submit($submittedValues, $action, $pledgePaymentID) {
       if (!empty($this->_payNow)) {
         $this->_params['contribution_id'] = $this->_id;
       }
+      // Since we are hiding the payment details block in edit mode,
+      // this also means that some payment information won't be there in
+      // submitted formValues. Since the BAO code does wrangling based on these
+      // it's important to ensure they are set. (ideally the BAO would cope
+      // with them being missing.
+      foreach (array(
+        'payment_instrument_id',
+        'check_number',
+        // ****** @Monish - is this suppose to be trxn_date!?
```

Comment:
So we can simply remove check_number. trxn_date is handled below. See at submit() function